addons/mail/models/discuss/discuss_channel_member.py

```diff
@@ -221,7 +221,59 @@ def write(self, vals):
             for field_name in ['channel_id', 'partner_id', 'guest_id']:
                 if field_name in vals and vals[field_name] != channel_member[field_name].id:
                     raise AccessError(_('You can not write on %(field_name)s.', field_name=field_name))
-        return super().write(vals)
+
+        def get_field_name(field_description):
+            if isinstance(field_description, Store.Attr):
+                return field_description.field_name
+            return field_description
+
+        def get_vals(member):
+            return {
+                get_field_name(field_description): (
+                    member[get_field_name(field_description)],
+                    field_description,
+                )
+                for field_description in self._sync_field_names()
+            }
+
+        old_vals_by_member = {member: get_vals(member) for member in self}
+        result = super().write(vals)
+        for member in self:
+            new_values = get_vals(member)
+            diff = []
+            for field_name, (new_value, field_description) in new_values.items():
+                old_value = old_vals_by_member[member][field_name][0]
+                if new_value != old_value:
+                    diff.append(field_description)
+            if diff:
+                diff.extend(
+                    [
+                        Store.One("channel_id", [], as_thread=True),
+                        *self.env["discuss.channel.member"]._to_store_persona([]),
+                    ]
+                )
+                if "message_unread_counter" in diff:
+                    # sudo: bus.bus: reading non-sensitive last id
+                    bus_last_id = self.env["bus.bus"].sudo()._bus_last_id()
+                    diff.append({"message_unread_counter_bus_id": bus_last_id})
+                Store(bus_channel=member._bus_channel()).add(member, diff).bus_send()
+        return result
+
+    @api.model
+    def _sync_field_names(self):
+        return [
+            "custom_channel_name",
+            "custom_notifications",
+            "last_interest_dt",
+            "message_unread_counter",
```

Comment:
Is it working for `message_unread_counter`? It's not a stored field so it should never be given in write?

Reply:
As discussed, the compute will change the value of this field before/after write when related values are changing (writing new_message_separator would trigger it for example).